src/rust/engine/process_execution/src/remote.rs

```diff
@@ -736,6 +753,30 @@ impl CommandRunner {
   }
 }
 
+fn maybe_add_workunit(result_cached: &bool, name: &str, start_time: &Timespec, end_time: &Timespec, parent_id: Option<String>, workunit_store: &Arc<WorkUnitStore>) {
+//  TODO: workunits for scheduling, fetching, executing and uploading should be recorded
+//   only if '--reporting-zipkin-trace-v2' is set
+  if !result_cached {
+    let workunit = WorkUnit {
+      name: String::from(name),
+      start_timestamp: timespec_as_float_secs(start_time),
+      end_timestamp: timespec_as_float_secs(end_time),
+      span_id: generate_random_64bit_string(),
+      parent_id,
+    };
+    workunit_store.add_workunit(workunit);
+  }
+}
+
+fn timespec_as_float_secs(timespec: &Timespec) -> f64 {
+  //  Returning value is formed by representing duration as a hole number of seconds (u64) plus
```

Comment:
Probably remove this comment - it's just describing what the code says and not adding much value on its own :)

Reply:
I'd probably keep the last line though, because we are losing precision and I can imagine myself forgetting about that very easily, and not getting that from the code.